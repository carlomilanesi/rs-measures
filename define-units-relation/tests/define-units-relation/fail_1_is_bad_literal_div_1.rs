rs_measures::define_measure_1d! {}

pub struct P1;

pub struct U1;
impl MeasurementUnit for U1 {
    type Property = P1;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u1";
}
impl VectorMeasurementUnit for U1 {}

pub struct P2;

pub struct U2;
impl MeasurementUnit for U2 {
    type Property = P2;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u2";
}
impl VectorMeasurementUnit for U2 {}

units_relation::define_units_relation! { U1 == 2 / U2 }

fn main() {}
