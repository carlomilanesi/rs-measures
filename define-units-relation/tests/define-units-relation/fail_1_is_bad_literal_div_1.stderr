error: expected identifier or literal `1`
  --> tests/define-units-relation/fail_1_is_bad_literal_div_1.rs:25:50
   |
25 | units_relation::define_units_relation! { U1 == 2 / U2 }
   |                                                  ^
